### PR TITLE
Baal hatanya

### LINKED ISSUE #65: admor hazaken
On 2017 Aug 18, at 17:09, ⁨אוהד אנטורג⁩ <⁨ant###@gmail.com⁩> wrote:

אפשר בבקשה לעדכן את השיטה של הזמנים על פי השיטה שמופיעה אתר זה? רוב חסידי חב"ד הולכים על פי שיטה זו 

https://sites.google.com/site/zmanimadmorhazaken/

Reported by: mwaisberg

### PR DESCRIPTION
**Pull request by yparitcher**
adds options according to the Baal Hatanya
fixes  #65

### CODE REVIEW

File: Halachic Times/app/src/main/res/values-he/strings.xml [hunk 1]
```diff
@@ -21,9 +21,10 @@
     <string name="opinion_96">יום הוא 96 דקות קודם הנץ / אחרי השקיעה</string>
     <string name="opinion_96_zmanis">יום הוא 96 דקות זמניות קודם הנץ / אחרי השקיעה</string>
     <string name="opinion_ateret">על דעת ישיבת עטרת תורה</string>
-    <string name="opinion_gra">על דעת הגר\"א ובעל התניא</string>
+    <string name="opinion_gra">על דעת הגרא</string>
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/res/values-he/strings.xml]: הגר\"א

File: Halachic Times/app/src/main/res/values-iw/strings.xml [hunk 2]
```diff
@@ -21,9 +21,10 @@
     <string name="opinion_96">יום הוא 96 דקות קודם הנץ / אחרי השקיעה</string>
     <string name="opinion_96_zmanis">יום הוא 96 דקות זמניות קודם הנץ / אחרי השקיעה</string>
     <string name="opinion_ateret">על דעת ישיבת עטרת תורה</string>
-    <string name="opinion_gra">על דעת הגר\"א ובעל התניא</string>
+    <string name="opinion_gra">על דעת הגרא</string>
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/res/values-iw/strings.xml]: הגר\"א

File: Halachic Times/app/src/main/res/values-lt/strings.xml [hunk 3]
```diff
@@ -211,5 +211,6 @@
     <string name="reminder_tuesday">Priminti antradien</string>
     <string name="reminder_wednesday">Priminti trečiadienį</string>
     <string name="notification_upcoming_title">Artėjantis laikas kaip pranešimo apie</string>
+    <string name="opinion_gra">Pasak GRA</string>
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/res/values-lt/strings.xml]: why did this move down here from line 24?

File: Halachic Times/app/src/main/java/com/github/times/ZmanimDetailsPopulater.java [hunk 4]
```diff
@@ -961,18 +1011,208 @@ private void populateLatestKiddushLevana(A adapter, ComplexZmanimCalendar cal, Z
...
-            case R.string.sunset:
-                populateSunset(adapter, cal, settings, offset);
-                break;
-            case R.string.twilight:
-                populateTwilight(adapter, cal, settings, offset);
-                break;
-            case R.string.nightfall:
-                populateNightfall(adapter, cal, settings, offset);
-                break;
+        Long date;
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/java/com/github/times/ZmanimDetailsPopulater.java]: please move all the "shabbat ends" to another pull request separate from the main Baal haTanya changes.

File: Halachic Times/app/src/main/res/values/arrays.xml [hunk 5]
```diff
@@ -372,14 +396,74 @@
     </string-array>
 
     <string-array name="shabbath_ends_after_values" translatable="false">
-        <item>@string/opinion_value_sunset</item>
-        <item>@string/opinion_value_twilight</item>
-        <item>@string/opinion_value_nightfall</item>
+        <item>@string/opinion_value_sea</item>
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/res/values/arrays.xml]: move another PR

File: Halachic Times/app/src/main/res/values/arrays.xml [hunk 6]
```diff
@@ -372,14 +396,74 @@
...
+        <item>@string/opinion_value_96_zmanis</item>
+        <item>@string/opinion_value_120</item>
+        <item>@string/opinion_value_120_zmanis</item>
+        <item>@string/opinion_value_26</item>
     </string-array>
     <string-array name="shabbath_ends_after_entries" translatable="false">
-        <item>@string/sunset</item>
-        <item>@string/twilight</item>
-        <item>@string/nightfall</item>
+        <item>@string/shabbath_ends_after_sea</item>
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/res/values/arrays.xml]: move another PR

File: Halachic Times/app/src/main/res/values/prefs.xml [hunk 7]
```diff
@@ -72,6 +72,7 @@
     <string name="opinion_value_default" translatable="false"/>
     <string name="opinion_value_fixed" translatable="false">fixed</string>
     <string name="opinion_value_gra" translatable="false">GRA</string>
+    <string name="opinion_value_baal_hatanya" translatable="false">baal_hatanya</string>
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/res/values/prefs.xml]: sorted alphabetically - please move it up to line 72.

File: Halachic Times/app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -25,14 +25,16 @@
     <string name="opinion_96">Day is 96 minutes before sunrise / after sunset</string>
     <string name="opinion_96_zmanis">Day is 96 זמניות minutes before sunrise / after sunset</string>
     <string name="opinion_ateret">According to Yeshivat Ateret Torah</string>
-    <string name="opinion_gra">According to GRA and Baal Tanya</string>
+    <string name="opinion_gra">According to GRA</string>
+    <string name="opinion_baal_hatanya">According to Baal Hatanya</string>
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/res/values/strings.xml]: sorted alphabetically

File: Halachic Times/app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -249,6 +262,40 @@
         <item quantity="other">@string/minutes_after_other</item>
     </plurals>
     <string name="festival_ends">Festival ends</string>
+    <string name="shabbath_ends_after_summary" translatable="false">@string/sunrise_summary</string>
```
Comment (pnemonic78) [inline Halachic Times/app/src/main/res/values/strings.xml]: move to another PR

File: Halachic Times/KosherJava/src/main/java/net/sourceforge/zmanim/ComplexZmanimCalendar.java [hunk 10]
```diff
@@ -2751,4 +2805,256 @@ public Long getSolarMidnight() {
...
+	 * less than 30 minutes {@link #getMinchaGedola30Minutes()} will be returned, otherwise {@link #getMinchaGedolaBaalHatanya()}
+	 * will be returned.
+	 *
+	 * @return the <code>Date</code> of the later of {@link #getMinchaGedolaBaalHatanya()} and {@link #getMinchaGedola30Minutes()}.
+	 *         If the calculation can't be computed such as in the Arctic Circle where there is at least one day a year
+	 *         where the sun does not rise, and one where it does not set, a null will be returned. See detailed
+	 *         explanation on top of the {@link AstronomicalCalendar} documentation.
+	 */
+	public Long getMinchaGedolaBaalHatanyaGreaterThan30() {
+		if (getTimeOffset(getSunriseBaalHatanya(), getShaahZmanisBaalHatanya() * 6.5) == null || getMinchaGedolaBaalHatanya() == null) {
```
Comment (pnemonic78) [inline Halachic Times/KosherJava/src/main/java/net/sourceforge/zmanim/ComplexZmanimCalendar.java]: please use local variables - each "getter" is potentially expensive.

File: Halachic Times/KosherJava/src/main/java/net/sourceforge/zmanim/ComplexZmanimCalendar.java [hunk 11]
```diff
@@ -2751,4 +2805,256 @@ public Long getSolarMidnight() {
...
+	 * {@link #ZENITH_6_DEGREES}.
+	 *
+	 * @return The <code>Date</code> of nightfall. If the calculation can't be computed such as northern and southern
+	 *         locations even south of the Arctic Circle and north of the Antarctic Circle where the sun may not reach
+	 *         low enough below the horizon for this calculation, a null will be returned. See detailed explanation on
+	 *         top of the {@link AstronomicalCalendar} documentation.
+	 * @see #ZENITH_6_DEGREES
+	 */
+	public Long getTzaisBaalHatanya() {
+		return this.getSunsetOffsetByDegrees(ZENITH_6_DEGREES);
```
Comment (pnemonic78) [inline Halachic Times/KosherJava/src/main/java/net/sourceforge/zmanim/ComplexZmanimCalendar.java]: don't need "this"

**Comment by yparitcher** [comment]
fixed

File: Halachic Times/app/src/main/res/values-lt/strings.xml (hunk 3 above)
Comment (yparitcher) [reply Halachic Times/app/src/main/res/values-lt/strings.xml]: andriod studio translation editor adds to the end, i didnt catch this one
